node/pod.go

```diff
@@ -213,8 +213,11 @@ func (pc *PodController) enqueuePodStatusUpdate(ctx context.Context, q workqueue
 		if obj, ok := pc.knownPods.Load(key); ok {
 			kpod := obj.(*knownPod)
 			kpod.Lock()
+			defer kpod.Unlock()
```

Comment:
This defer could keep `kpod` locked if `AddRateLimited` is blocked.
I would prefer if we play this safer and just unblock immediately after `cmp.Equal`

Reply:
Seems safer, done!